src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -124,80 +106,50 @@ public boolean compile() throws IOException, InterruptedException {
     public boolean link() throws IOException, InterruptedException {
         // we override link as we need to do some checks first. If we have no clang in android_ndk, we should not start linking
         if (ndk == null) throw new IOException ("Can't find an Android NDK on your system. Set the environment property ANDROID_NDK");
-        if (clang == null) throw new IOException ("You specified an android ndk, but it doesn't contain "+ndk+"/toolchains/llvm/prebuilt/"+hostPlatformFolder+"/bin/clang");
+        if (clang == null) throw new IOException ("You specified an android NDK, but it doesn't contain "+ndk+"/toolchains/llvm/prebuilt/"+hostPlatformFolder+"/bin/clang");
         if (sdk == null) throw new IOException ("Can't find an Android SDK on your system. Set the environment property ANDROID_SDK");
 
         return super.link();
     }
 
     @Override
     public boolean packageApp() throws IOException, InterruptedException {
-        Path androidPathForManifest = prepareAndroidManifest();
-        Path androidPathForRes = prepareAndroidResources();
-
-        ensureApkOutputDirectoriesExist();
-
-        Path sdkPath = Paths.get(sdk);
-        Path buildToolsPath = sdkPath.resolve("build-tools").resolve(findLatestBuildTool(sdkPath));
-        String androidJar = sdkPath.resolve("platforms").resolve("android-27").resolve("android.jar").toString();
-
-        String unalignedApk = getApkBinPath().resolve(projectConfiguration.getAppName() + ".unaligned.apk").toString();
-        String alignedApk = getApkBinPath().resolve(projectConfiguration.getAppName() + ".apk").toString();
-
-        if (!processPrecompiledClasses(androidJar)) {
-            return false;
-        }
-
-        copyAndroidManifest(androidPathForManifest);
-        copyAssets(androidPathForRes);
-
-        int processResult = dx(buildToolsPath);
-        if (processResult != 0) {
-            return false;
-        }
-
-        processResult = aapt(buildToolsPath, unalignedApk, androidJar);
-        if (processResult != 0) {
-            return false;
-        }
-
-        processResult = zipAlign(buildToolsPath, unalignedApk, alignedApk);
-        if (processResult != 0) {
-            return false;
-        }
+        prepareAndroidProject();
+        prepareAndroidManifest();
+        prepareAndroidResources();
+        copyOtherDalvikClasses();
+        copySubstrateLibraries();
 
-        processResult = sign(buildToolsPath, alignedApk);
-        return processResult == 0;
+        ProcessRunner assembleDebug = new ProcessRunner(
```

Comment:
I take that `assembleDebug` will sign internally the apk for debug, and the `debug.keystore` we created is no longer required, right?
What about signing for release? I see that you have removed the `sign` method. What is the alternative? Could we create the signing configuration in the build.gradle file, with the release data?

Reply:
This should be fixed in latest commit